pkg/ui/src/views/databases/containers/databaseSummary/index.tsx

```diff
@@ -44,45 +45,55 @@ interface DatabaseSummaryActions {
   refreshTableStats: typeof refreshTableStats;
 }
 
-type DatabaseSummaryProps = DatabaseSummaryExplicitData & DatabaseSummaryConnectedData & DatabaseSummaryActions;
+export type DatabaseSummaryProps = DatabaseSummaryExplicitData & DatabaseSummaryConnectedData & DatabaseSummaryActions;
+
+interface DatabaseSummaryState {
+  finishedLoadingTableData: boolean;
+}
 
 // DatabaseSummaryBase implements common lifecycle methods for DatabaseSummary
 // components, which differ primarily by their render() method.
 // TODO(mrtracy): We need to find a better abstraction for the common
 // "refresh-on-mount-or-receiveProps" we have in many of our connected
 // components; that would allow us to avoid this inheritance.
-export class DatabaseSummaryBase extends React.Component<DatabaseSummaryProps, {}> {
+export class DatabaseSummaryBase extends React.Component<DatabaseSummaryProps, DatabaseSummaryState> {
   // loadTableDetails loads data for each table which have no info in the store.
   // TODO(mrtracy): Should this be refreshing data always? Not sure if there
   // is a performance concern with invalidation periods.
-  loadTableDetails(props = this.props) {
+  async loadTableDetails(props = this.props) {
     if (props.tableInfos && props.tableInfos.length > 0) {
-      _.each(props.tableInfos, (tblInfo) => {
+      for (const tblInfo of props.tableInfos) {
+        // TODO(davidh): this is a stopgap inserted to deal with DBs containing hundreds of tables
         if (_.isUndefined(tblInfo.numColumns)) {
-          props.refreshTableDetails(new protos.cockroach.server.serverpb.TableDetailsRequest({
+          await props.refreshTableDetails(new protos.cockroach.server.serverpb.TableDetailsRequest({
```

Comment:
Would it be reasonable to have two parallel calls inside the loop? Currently `refreshTableDetails` and `refreshTableStats` calls are sequential and might increase response time for a single table.

As an option it could be called in parallel:
```typescript
await Promise.all([
  props.refreshTableDetails(new protos.cockroach.server.serverpb.TableDetailsRequest({
    database: props.name,
    table: tblInfo.name,
  })),
  props.refreshTableStats(new protos.cockroach.server.serverpb.TableStatsRequest({
    database: props.name,
    table: tblInfo.name,
  })),
])
```

PS: on a local dev environment, with ~35 tables in a database, parallel requests complete entire loading a little faster. 
Sequential requests: 15503.633056640625 ms
Parallel requests (with Promise.all): 14526.049072265625 ms